src/llmcompressor/transformers/finetune/text_generation.py

```diff
@@ -432,20 +466,62 @@ def main(
         stage_runner.predict()
 
     # save if model was provided as a string or custom output_dir was set
+
     if isinstance(model_args.model, str) or (
         training_args.output_dir
         != TrainingArguments.__dataclass_fields__["output_dir"].default
     ):
         model.save_pretrained(
-            training_args.output_dir, save_compressed=training_args.save_compressed
+            training_args.output_dir, save_compressed=model_args.save_compressed
         )
         if processor is not None:
             processor.save_pretrained(training_args.output_dir)
 
     # Clean up the CompressionSession before exit if requested
-    if training_args.clear_sparse_session:
+    if recipe_args.clear_sparse_session:
         reset_session()
 
 
+def _validate_model_args_tokenizer(model_args: ModelArguments):
+    """Ensure only one of tokenizer or processor is set"""
+
+    if model_args.tokenizer:
+        if model_args.processor:
+            raise ValueError("Cannot use both a tokenizer and processor.")
+        model_args.processor, model_args.tokenizer = model_args.tokenizer, None
+
+
+def _get_output_dir_from_argv() -> Optional[str]:
+    """Extract output directory from command-line arguments"""
+
+    import sys
+
+    if "--output_dir" in sys.argv:
+        index = sys.argv.index("--output_dir")
+        sys.argv.pop(index)
+        if index < len(sys.argv):
+            return sys.argv.pop(index)
+
+    return None
+
+
+def _get_dataclass_arguments(include_training_args: bool):
+    """Return the appropriate argument classes for parsing"""
+
+    dataclass_arguments = (ModelArguments, DatasetArguments, RecipeArguments)
+    if include_training_args:
+        return dataclass_arguments + (TrainingArguments,)
+
+    return dataclass_arguments
+
+
+def _unwrap_recipe_args(recipe_args: Dict[str, Any]):
+    """Convert recipe arguments to a dictionary if needed"""
+    if isinstance(recipe_args, dict):
+        return recipe_args
+
+    return {key: value for arg in recipe_args for key, value in [arg.split("=")]}
```

Comment:
This is overly complicated and kind of misleading.
```
{
    key: value
    for arg in recipe_args
    for key, value in [arg.split("=")]
}
```
Assuming that `arg.split("=")` only yields 2 values (which is the previous assumption), we get a list [(key, value)]. Technically you could use `for` to iterate through this singleton list to unpack it, but doing so is kind of misleading and unnecessary.

If you had to do it as a list comprehension, you could do it like this.
```suggestion
    {
        key: value
        for key, value in (arg.split("=") for arg in training_args.recipe_args)
    }
```

But in my opinion, the original for loop is the clearest.

Side note: Even if this code was written with the intention of supporting multiple kv pairs, it would yield [(k1, v1, k2, v2)] which does not unpack correctly.

Reply:
I dont understand.